utils/jvm_dependencies.py

```diff
@@ -0,0 +1,131 @@
+import os
+import sys
+import tempfile
+import requests
+import hashlib
+import subprocess
+from dataclasses import dataclass
+
+
+@dataclass
+class Artifact:
+    group: str
+    name: str
+    version: str
+    url: str
+    bazelized_name: str
+
+
+def parse_line(line: str) -> Artifact:
+    parts = line.strip().split(':')
+    if len(parts) == 4:
+        group, name, version, _ = parts
+        group_path = group.replace('.', '/')
+        jar_name = f"{name}-{version}.jar"
+        url = f"https://repo1.maven.org/maven2/{group_path}/{name}/{version}/{jar_name}"
+        bazelized_name = name.replace('-', '_').replace('.', '_')
+        return Artifact(group, name, version, url, bazelized_name)
+
+
+def download_and_compute_sha256(artifact: Artifact) -> str:
+    # Create a temporary directory for the downloaded file
+    temp_dir = tempfile.mkdtemp()
+    file_path = os.path.join(temp_dir, artifact.name + '.jar')
+
+    try:
+        # Download the artifact
+        response = requests.get(artifact.url)
+        response.raise_for_status()
+
+        # Save the downloaded file
+        with open(file_path, 'wb') as file:
+            file.write(response.content)
+
+        # Compute the SHA-256 hash
+        sha256_hash = hashlib.sha256()
+        with open(file_path, 'rb') as file:
+            while True:
+                data = file.read(65536)  # Read in 64k chunks
+                if not data:
+                    break
+                sha256_hash.update(data)
+
+        print(f"Downloaded {artifact.name}", file=sys.stderr)
```

Comment:
so crazy that sonatype can't provide this data from the registry without having to download :(

Reply:
and [repository_ctx.download](https://bazel.build/rules/lib/builtins/repository_ctx#download) accepts only sha256

Reply:
I don't think that's true, you should be able to use the `integrity` attribute instead, like `integrity="sha1:[base64-encoded value]"`
There's a `base64` function in aspect bazel-lib.
If this is too much extra effort I'm okay with leaving a TODO here instead...

Reply:
ahh, right, I didn't notice it and never saw anyone using this attribute

Reply:
I'll fix it later today

Reply:
nice, if you get time that's great. Thank you for all your effort!

Reply:
Done :)